src/neptune/legacy/__init__.py

```diff
@@ -76,6 +77,7 @@ def _check_for_extra_kwargs(caller_name, kwargs: dict):
         raise TypeError(f"{caller_name}() got an unexpected keyword argument '{first_key}'")
 
 
+@legacy_client_deprecation
```

Comment:
If we're decorating all the functions in the module, why not just throw a warning on the top of the module so it will be displayed at import from it?

Reply:
Most of them are imported at root module `__init__.py` (https://github.com/neptune-ai/neptune-client/blob/9a0e977b8345cdba93d3c4d34547591e7649e03e/src/neptune/__init__.py#L17) so it may impact either `neptune.new` & `neptune.management`.